### DISCUSSION TITLE
Python version upgrade

### DISCUSSION THREAD

**Discussion by c-martinez** [CLARIAH/grlc#409] [Polls]
Grlc currently supports Python 3.7, 3.8 and 3.9. but 3.7 has [reached its end of life](https://devguide.python.org/versions/), with 3.11 being the current stable version.

**Comment by c-martinez** [comment]
Depending on the outcome of this, we should update dependencies such as Flask (#399)

**Comment by RinkeHoekstra** [comment]
Hi Carlos, all,

I think the key question is: what vulnerabilities exist in v3.7 libraries that grlc is using? In general, I'd be in favour of tracking at least the latest stable version of each library, and all security fixes. If that requires an update of the Python version, then so be it. We can't have vulnerabilities linger for too long.

One thing to consider is that AWS linux comes with Python 3.9 out of the box (but does offer 3.11 binaries as of this year).

Great to see that `grlc` is still alive and kicking!

**Comment by c-martinez** [comment]
Unless there are any objections, we will go for supporting 3.11,3.10, 3.9 (and 3.8 if possible & until next year)
